mne_bids/mne_bids.py

```diff
@@ -158,6 +158,68 @@ def _events_tsv(events, raw, fname, trial_type, verbose):
     return fname
 
 
+def _participants_tsv(raw, group, fname, verbose):
+    """Create a participants.tsv file and save it.
+
+    This will append any new participant data to the current list if it
+    exists
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The data as MNE-Python Raw object.
+    group : str
+        Name of group participant belongs to.
+    fname : str
+        Filename to save the participants.tsv to.
+    verbose : bool
+        Set verbose output to true or false.
+
+    """
+    subject_info = raw.info['subject_info']
+    if subject_info is not None:
+        genders = {0: 'U', 1: 'M', 2:'F'}
+        sex = genders[subject_info.get('sex', 0)]
+        # this will have issues if the his_id doesn't actually match the
+        # subject id provided...
+        subject_id = 'sub-' + subject_info['his_id']
+        try:
+            age = subject_info['birthday']
+            bday = datetime(age[0], age[1], age[2])
+            meas_date = raw.info['meas_date']
+            if isinstance(meas_date, (np.ndarray, list)):
+                meas_date = meas_date[0]
+            meas_date = datetime.fromtimestamp(meas_date)
+            subject_age = age_on_date(bday, meas_date)
+        except KeyError:
```

Comment:
avoid `try-catch` when you can. Why don't you use the `.get()` thing that you were using elsewhere?

Reply:
I thought the logic wouldn't be as nice since there are two spots it can be broken at, but it doesn't look too bad:
```python
        age = subject_info.get('birthday', None)
        meas_date = raw.info.get('meas_date', None)
        if meas_date is not None and age is not None:
            bday = datetime(age[0], age[1], age[2])
            if isinstance(meas_date, (np.ndarray, list)):
                meas_date = meas_date[0]
            meas_date = datetime.fromtimestamp(meas_date)
            try:
                subject_age = age_on_date(bday, meas_date)
            except ValueError:
                subject_age = "n/a"
        else:
            subject_age = "n/a"
```
However, I just realised that we should probably catch the case of a ValueError being raised by the `age_on_date` function, which make the above a bit less nice, but for the current code you can just do `except (KeyError, ValueError):` and be done with it.